Calibration/HcalAlCaRecoProducers/plugins/AlCaHBHEMuonFilter.cc

```diff
@@ -182,26 +202,35 @@ bool AlCaHBHEMuonFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSetu
     }
   }
   // Step 4:  Return the acceptance flag
-  if (accept) nGood_++;
+  if (accept) ++nGood_;
   return accept;
 
 }  // AlCaHBHEMuonFilter::filter
 
 // ------------ method called once each job just after ending the event loop  ------------
-void AlCaHBHEMuonFilter::endJob() {
-  edm::LogInfo("HcalHBHEMuon") << "Selects " << nGood_ << " in " << nAll_ 
-			       << " events from " << nRun_ << " runs";
-  for (unsigned int k=0; k<trigNames_.size(); ++k)
-    edm::LogInfo("HcalHBHEMuon") << "Trigger[" << k << "]: " << trigNames_[k] 
-				 << " Events " << trigKount_[k] << " Passed " 
-				 << trigPass_[k];
+void AlCaHBHEMuonFilter::endStream() {
+  globalCache()->nAll_  += nAll_;
+  globalCache()->nGood_ += nGood_;
+  for (unsigned int k=0; k<trigNames_.size(); ++k) {
+    globalCache()->trigKount_[k] += trigKount_[k];
```

Comment:
Since multiple endStreams could be called simultaneously, this is not thread safe.